artemis-commons/src/main/java/org/apache/activemq/artemis/utils/ActiveMQThreadPoolExecutor.java

```diff
@@ -46,32 +45,70 @@
    private static class ThreadPoolQueue extends LinkedBlockingQueue<Runnable> {
 
       private ActiveMQThreadPoolExecutor executor = null;
+      private int idleThreads = 0;
 
       public void setExecutor(ActiveMQThreadPoolExecutor executor) {
          this.executor = executor;
       }
 
       @Override
-      public boolean offer(Runnable runnable) {
+      public synchronized boolean offer(Runnable runnable) {
```

Comment:
@franz1981 I feel like you're the man to look into this...

can you look into this one?